public/src/inverted-index.js

```diff
@@ -92,63 +101,57 @@ class Index {
     return check;
   }
 
-/**
- * @method getIndex
- *
- * Method return index of files in indices object
- *
- * @param {string} name
- * @returns {object}
- */
-  getIndex(name) {
-    if (name && typeof name === 'string') {
-      return this.indices[name];
-    }
-    return this.indices;
-  }
+  /**
+  * @method getIndex
+  *
+  * Method return index of files in indices object
+  *
+  * @param {string} name
+  * @returns {object}
+  */
+  getIndex(name) { return name && typeof name === 'string' ? this.indices[name] : this.indices; }
 
-/**
- * @method searchFeedback
- *
- * Takes terms of array and fetch result of each token.
- *
- * @param {array} termsArray
- * @param {object} file
- * @returns {object} searchResults
- */
-  searchFeedback(termsArray, file) {
+  /**
+  * @method doSearch
+  *
+  * Takes terms of array and fetch result of each token.
+  *
+  * @param {array} termsArray
+  * @param {object} file
+  * @returns {object} searchResults
+  */
+  doSearch(termsArray, fileIndex) {
     const searchResults = {};
     termsArray.forEach((term, index) => {
-      if (file.hasOwnProperty(term)) {
-        searchResults[termsArray[index]] = Array.from(file[term]);
+      if (fileIndex.hasOwnProperty(term)) {
+        searchResults[termsArray[index]] = Array.from(fileIndex[term]);
       } else {
         searchResults[term] = [];
       }
     });
     return searchResults;
   }
 
-/**
- * @method searchIndex
- *
- * Looks for search terms in created index
- *
- * @param {object} currentFile
- * @param {array} searchTerm
- * @returns {object} searchResults
- */
+  /**
+  * @method searchIndex
+  *
+  * Looks for search terms in created index
+  *
+  * @param {object} currentFile
+  * @param {array} searchTerm
+  * @returns {object} searchResults
+  */
   searchIndex(currentFile, ...searchTerm) {
     const searchResults = {};
-    let termsArray = [];
-    termsArray = this.sanitizeInput(searchTerm[0]);
+    const termsArray = this.sanitizeInput(searchTerm[0]);
     if (!currentFile) {
```

Comment:
### Critical

This assertion would never be TRUE because for instance, when `searchIndex('alice', ['hello', 'world'])` is called, it assumes `alice` is the filename(of which it isn't) i.e filename is provided, thus making this method always default to the `else` block. 

You need to assert the first argument is in fact a filename/has a corresponding index.